relevanceai/base.py

```diff
@@ -16,3 +18,19 @@ def __init__(self, project: str, api_key: str):
         self.config = CONFIG
         # Initialize logger
         super().__init__()
+
+
+class ClusterBase:
+    def __init__(self, clusterer):
+        self.clusterer = clusterer
+
+    @abstractmethod
+    def fit_transform(self, vectors):
+        raise NotImplementedError
+
+    @abstractmethod
+    def get_centroids(self):
+        raise NotImplementedError
+
+    def metadata(self):
+        return self.clusterer.__dict__

```

Comment:
This requires a `hasattr`. If it does not have the `__dict__` attribute, this should not return the right thing.